tests/test_models.py

```diff
@@ -0,0 +1,237 @@
+
+    def test_succeed(self):
+        """succeed() should update its parent only if all its other children have already succeeded."""
+        parent = self._status(is_container=True, total_steps=10)
+        child1 = self._status(parent=parent)
+        child2 = self._status(parent=parent)
+        child1.succeed()
+        parent.refresh_from_db()
+        assert child1.state == UserTaskStatus.SUCCEEDED
+        assert child1.completed_steps == child1.total_steps
+        assert parent.state == UserTaskStatus.IN_PROGRESS
+        assert parent.completed_steps == child1.total_steps
+        child2.succeed()
+        parent.refresh_from_db()
+        assert child2.state == UserTaskStatus.SUCCEEDED
+        assert child2.completed_steps == child2.total_steps
+        assert parent.state == UserTaskStatus.SUCCEEDED
+        assert parent.completed_steps == child1.total_steps + child2.total_steps
+
+    def _status(self, **kwargs):
+        """Generate a sample UserTaskStatus instance, optionally overriding fields with keyword arguments."""
+        data = {
+            'name': 'SampleTask', 'state': UserTaskStatus.IN_PROGRESS, 'task_class': 'test_models.sample_task',
+            'task_id': str(uuid4()), 'total_steps': 5, 'user': self.user}
+        data.update(kwargs)
+        return UserTaskStatus.objects.create(**data)
+
+
+class TestUserTaskArtifact(TestCase):
+    """Tests of the UserTaskArtifact model."""
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.user = User.objects.create_user('test_user', 'test@example.com', 'password')
+
+    def test_file_string(self):
+        """File-based UserTaskArtifacts should have a reasonable string representation"""
+        status = self._status()
+        file = SimpleUploadedFile('file.txt', b'File content')
+        artifact = UserTaskArtifact(status=status, file=file)
+        assert str(artifact) == '<UserTaskArtifact: (Output) file.txt>'
+
+    def test_text_string(self):
+        """Text-based UserTaskArtifacts should have a reasonable string representation."""
+        status = self._status()
+        artifact = UserTaskArtifact(status=status, text='This was the output text.')
+        assert str(artifact) == '<UserTaskArtifact: (Output) This was the output text.>'
+
+    def test_long_text_string(self):
+        """Long text artifacts should be cropped for the string representation."""
+        status = self._status()
+        text = 'blah ' * 50
+        artifact = UserTaskArtifact(status=status, name='Cropped', text=text)
+        assert str(artifact) == '<UserTaskArtifact: (Cropped) {}>'.format(text[:47] + '...')
+
+    def test_url_string(self):
+        """URL-based UserTaskArtifacts should have a reasonable string representation."""
+        status = self._status()
+        artifact = UserTaskArtifact(status=status, url='http://www.edx.org/')
+        assert str(artifact) == '<UserTaskArtifact: (Output) http://www.edx.org/>'
+
+    def _status(self, **kwargs):
```

Comment:
Could, but then I'd have to explicitly pass in the user as an argument for every invocation instead of plucking it from `self`.